src/write_gcm.c

```diff
@@ -2275,59 +2274,155 @@ static monitored_resource_t *monitored_resource_create_from_fields(
   // Changes va_list into label_t array
   va_start(ap, project_id);
   for (int i = 0; i < num_labels; i++) {
-    labels[i].key = va_arg(ap, const char*);
-    labels[i].value = va_arg(ap, const char*);
+    labels[i].key = va_arg(ap, char*);
+    labels[i].value = va_arg(ap, char*);
   }
   va_end(ap);
   return monitored_resource_create_from_array(type, project_id, labels,
                                               num_labels);
 }
 
+typedef struct label_node_s
+{
+  label_t label;
+  struct label_node_s *next;
+} wg_mr_label_node_t;
+
+typedef struct {
+  int processing_labels;
+  int total_labels;
+  char *last_key;
+  char *type;
+  wg_mr_label_node_t *head;
+} wg_mr_parse_context_t;
+
+static int wg_mr_handle_string(void * arg, const unsigned char * val,
+    wg_yajl_callback_size_t length) {
+  wg_mr_parse_context_t *ctx = (wg_mr_parse_context_t*) arg;
+
+  if(ctx->processing_labels) {
+    label_t label = {
+      .key = ctx->last_key,
+      .value = strndup((const char*) val, length)
+    };
+
+    wg_mr_label_node_t *label_node = (wg_mr_label_node_t*) calloc(1, sizeof(wg_mr_label_node_t));
+    label_node->label = label;
```

Comment:
I would have suggested something like
```C
  wg_mr_label_node_t *label_node = (wg_mr_label_node_t*) calloc(1, sizeof(wg_mr_label_node_t));
  label_node->key = ctx->last_key;
  label_node->value = strndup((const char*) val, length);
```
But it would be even simpler to do the following:
```C
  llist_append(ctx->label_list, llentry_create(ctx->last_key, strndup((const char*) val, length)));
```
Then you can also get rid of [lines 2312-2315](https://github.com/Stackdriver/collectd/blob/e7557a0c008d0f89ff9aa6d2a47fc1d4eecc9bdb/src/write_gcm.c#L2312-L2315).